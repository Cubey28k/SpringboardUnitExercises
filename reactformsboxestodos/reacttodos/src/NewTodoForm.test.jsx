import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewToDoForm from "./NewTodoForm";

it("renders without crashing", function() {
    render(<NewToDoForm />);
});

it("matches snapshot", function() {
    const { asFragment } = render(<NewToDoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("runs the create function on form submit", function() {
    const createMock = jest.fn();
    const { getByText } = render(<NewToDoForm createTodo={createMock} />);
    const createButton = getByText("Add a todo!");
    fireEvent.click(createButton);
    expect(createMock).toHaveBeenCalled();
});