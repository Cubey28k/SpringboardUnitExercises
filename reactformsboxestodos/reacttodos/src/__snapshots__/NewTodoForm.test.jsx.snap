// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Add a todo!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
